lib/ansible/config/manager.py

```diff
@@ -148,6 +148,9 @@ def ensure_type(value, value_type, origin=None):
             else:
                 errmsg = 'string'
 
+        elif value_type == 'raw':
+            value = to_text(value, errors='surrogate_or_strict')
```

Comment:
i would not use to_text here as 'raw' is meant to bypass any conversion at the generic code and leave for plugin to handle